### PR TITLE
move to immutable state

### PR DESCRIPTION
**Pull request by VictorUvarov**
- update pedantic to fix lint warnings

### CODE REVIEW

File: lib/state/base_view_model.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/widgets.dart';
+
+class BaseViewModel<T> extends ChangeNotifier {
```
Comment (bsneider) [inline lib/state/base_view_model.dart:3]: Why not use the base from Stacked itself?

File: lib/ui/views/home/home_view_model.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import 'package:provider_start/core/repositories/posts_repository/posts_repository.dart';
+import 'package:provider_start/locator.dart';
+import 'package:provider_start/state/base_view_model.dart';
+import 'package:provider_start/ui/views/home/home_view_state.dart';
+
+class HomeViewModel extends BaseViewModel<HomeViewState> {
+  final _postsRepository = locator<PostsRepository>();
+
+  Future<void> init() async {
+    await _attemptFetchPosts();
```
Comment (bsneider) [inline lib/ui/views/home/home_view_model.dart:11]: Checkout FutureViewModel, its pretty cool and can do the async really nicely

File: lib/state/base_view_model.dart (hunk 1 above)
Comment (VictorUvarov) [reply lib/state/base_view_model.dart:3]: It wasn't designed to have immutable state. I'll look into it more.

File: lib/ui/views/home/home_view_model.dart (hunk 2 above)
Comment (VictorUvarov) [reply lib/ui/views/home/home_view_model.dart:11]: Thanks I'll check it out, not sure if I can mix the view model with my own.

**Pull request merged**